### PR TITLE
[Compute] `az sig image-version`: Add new parameter `--replication-mode` to support setting replication mode

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
Feature request: #18851

**Description**

Add a feature named "replication mode" under the image-definition object in Shared Image Gallery. 

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -3566,10 +3567,19 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
...
+                                                target_regions=target_regions or [TargetRegion(name=location)],
+                                                source=source, replica_count=replica_count,
+                                                storage_account_type=storage_account_type,
+                                                replication_mode=replication_mode)
+    else:
+        profile = ImageVersionPublishingProfile(exclude_from_latest=exclude_from_latest,
+                                                end_of_life_date=end_of_life_date,
+                                                target_regions=target_regions or [TargetRegion(name=location)],
+                                                source=source, replica_count=replica_count,
+                                                storage_account_type=storage_account_type)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:3582]: May I ask why is the parameter `replication_mode` not allowed to be passed into `None` for `ImageVersionPublishingProfile`?

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -1055,6 +1055,7 @@ def load_arguments(self, _):
         c.argument('data_vhds_uris', nargs='+', help='Source VHD URIs (space-delimited) of data disks')
         c.argument('data_vhds_luns', nargs='+', help='Logical unit numbers (space-delimited) of source VHD URIs of data disks')
         c.argument('data_vhds_storage_accounts', options_list=['--data-vhds-storage-accounts', '--data-vhds-sa'], nargs='+', help='Names or IDs (space-delimited) of storage accounts of source VHD URIs of data disks')
+        c.argument('replication_mode', min_api='2021-07-01', arg_type=get_enum_type(['Full', 'Shallow']), help='Optional parameter which specifies the mode to be used for replication. This property is not updatable.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1058]: May I ask ae these enumeration values defined in the Python SDK? If so, please refer to the enumeration values from the Python SDK

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (PARADISSEEKR) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:3582]: If the parameter passed in `sreplication_mode` is `None`, the return result is `null`, which is the same as the result without passing in parameters.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 2 above)
Comment (PARADISSEEKR) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:1058]: It is the same as the definition in the SDK.

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:1058]: Please use the definitions from SDK

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:3582]: How about
```Python
profile = ImageVersionPublishingProfile(exclude_from_latest=exclude_from_latest,
                                                end_of_life_date=end_of_life_date,
                                                target_regions=target_regions or [TargetRegion(name=location)],
                                                source=source, replica_count=replica_count,
                                                storage_account_type=storage_account_type)
 if replication_mode is not None:
      profile.replication_mode = replication_mode 
```

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@PARADISSEEKR Could you address those conflicts?

**Pull request merged**